Lib/configparser.py

```diff
@@ -1048,6 +1052,38 @@ def __iter__(self):
         # XXX does it break when underlying container state changed?
         return itertools.chain((self.default_section,), self._sections.keys())
 
+    def __str__(self):
+        config_dict = {
+            section: dict(self.items(section, raw=True))
+            for section in self.sections()
+        }
+        return f"<ConfigParser: {config_dict}>"
+
+    def __repr__(self):
+        init_params = {
+            "defaults": self._defaults if self._defaults else None,
+            "dict_type": type(self._dict).__name__,
+            "allow_no_value": self._allow_no_value,
+            "delimiters": self._delimiters,
+            "strict": self._strict,
+            "default_section": self.default_section,
+            "interpolation": type(self._interpolation).__name__,
+        }
+        init_params = {k: v for k, v in init_params.items() if v is not None}
+        sections_count = len(self._sections)
+        state_summary = {
+            "loaded_sources": self._loaded_sources,
+            "sections_count": sections_count,
+            "sections": list(self._sections)[:5],  # limit to 5 section names for readability
+        }
+
+        if sections_count > 5:
+            state_summary["sections_truncated"] = f"...and {sections_count - 5} more"
+
+        return (f"<{self.__class__.__name__}("
+                f"params={init_params}, "
+                f"state={state_summary})>")
```

Comment:
We could use `=` f-strings:

```py
return f"<{self.__class__.__name__}({params=}, {state=})>"
```

but this requires the variables to be named "params" and "state" instead.

Reply:
I can change if you want but `init_params` and `state_summary` are more descriptive.

Reply:
Mmh, yes. Let's leave keep it that way (I'll let Jaraco have the final say)